### ISSUE TITLE
Enable Hot Push Code

### ISSUE THREAD

**Issue by chauthai** [eduedix/react-native-networking#1]
Since you enable downloading new content from an external source to the app, I wonder if you can update the main.jsbundle with this approach.

You define the location of the main.jsbundle in the AppDelegate.m. So I imagine if you trigger a download you may be able to update it? But I guess the current location of main.jsbundle is in a secure spot, so you have to point the AppDelegate.m to your existing Documents folder. In some cases that might be security problem.

As an inspiration two plugins to enable hot pushing code in Cordova:
https://github.com/markmarijnissen/cordova-app-loader

https://github.com/mathieudutour/cordova-plugin-hotpushes based on
https://github.com/phonegap/phonegap-plugin-contentsync

Both plugins check for new versions via a json file on the host server and only trigger updates when there is a newer version. Also the contentsync plugin provides different states do identify the current download status.

TL;DR: Enable hot push code to the app:
- version check of main.jsbundle
- download new main.jsbundle
- replace old main.jsbundle / change location of main.jsbundle in AppDelegate.m to reflect update 
- reload app when update is done to roll out updated main.jsbundle

**Comment by eduedix** [comment]
This should be done in another module.
